doc/sailfish-webview-webview.qdoc

```diff
@@ -123,26 +612,191 @@
 */
 
 /*!
-  \qmlproperty qreal WebView::virtualKeyboardMargin
-  \brief Set the margin taken by the keyboard at the bottom of the page
+  \qmlmethod WebView::loadHtml(string html, url baseUrl)
+  \brief Loads the specified \a html content, performing symbol resolution
+         from the given \a baseUrl
+*/
 
-  When the keybaord is opened it may overlap web content displayed in a
-  \l{WebView}. Setting this value allows the web content to be arranged to
-  compensate for the overlapped area.
+/*!
+  \qmlmethod WebView::loadText(string text, string mimeType)
+  \brief Loads the specified \a text content, displaying appropriately for
+         the specified \a mimeType
+*/
 
-  When using a \l{WebView} this property will be set automatically when the
-  keybaord is opened or closed.
+/*!
+  \qmlmethod WebView::goBack()
+  \brief Navigate back to the previous location in history
 */
 
 /*!
-  \qmlsignal WebView::contentOrientationChanged(Qt::ScreenOrientation orientation)
-  \brief This signal is emitted when the screen orientation of the rendered
-         content changes.
+  \qmlmethod WebView::goForward()
+  \brief Navigate forward to the next location in history
+*/
+
+/*!
+  \qmlmethod WebView::stop()
+  \brief Stop loading the current URL or content
+*/
+
+/*!
+  \qmlmethod WebView::reload()
+  \brief Reload the current URL or content
+*/
+
+/*!
+  \qmlmethod WebView::load(string url)
+  \brief Loads the specified \a url
+*/
+
+/*!
+  \qmlmethod WebView::sendAsyncMessage(string name, variant data)
+  \brief Send an asynchronous message with the specified \a name and \a data
+
+  If a message listener is registered for the message, it will handle it
+  appropriately (e.g. show an appropriate popup, etc).
+
+  \sa addMessageListener()
+*/
+
+/*!
+  \qmlmethod WebView::addMessageListener(string name)
+  \brief Register a listener for asynchronous messages with the
+         specified \a name
+*/
+
+/*!
+  \qmlmethod WebView::loadFrameScript(string name)
+  \brief Loads the specified frame script
```

Comment:
A frame script is javascript that is loaded for the webview to the engine "below" the actually loaded web content i.e. frame script has higher rights to manipulate the web content (chrome privileged / system principal).

Chrome is registered uri scheme and for instance embedhelper.js of the embedlite-components is loaded for each created view (chrome://embedlite/content/embedhelper.js). <= We do not need to advertise this.

My app structure:
```
<path>/my-app/pages/HelloWorld.qml
<path>/my-app/pages/myframescript.js
```
=>
```
webView.loadFrameScript(Qt.resolvedUrl("myframescript.js"));
webView.addMessageListener("myappmessagehandler")
```

The myframescript.js could for instance... addEventListener of a kind, do some handling, and sendAsyncMessage("myappmessagehandler", {"somedata": 2}) back to the app.

Reply:
thanks, added some info to the doc